aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/AWSAppSyncDeltaSyncSqlHelper.java

```diff
@@ -33,9 +35,15 @@
             COLUMN_DELTA_SYNC_KEY,
             COLUMN_LAST_RUN_TIME);
 
+    synchronized public static AWSAppSyncDeltaSyncSqlHelper create(Context context) {
```

Comment:
You've only made one constructor `private`, but there are still others that allow instances to be made. So, this does  not actually enforce a singleton pattern.